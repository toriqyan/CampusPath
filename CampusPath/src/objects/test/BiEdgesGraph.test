# Tests that a graph can be created, four nodes and
# two pairs of cross edges and two other edges can be added, 
# and four nodes and six edges can be printed out.

# Create a graph
CreateGraph TheLordOfTheRing5

# Add four nodes
AddNode TheLordOfTheRing5 Frodo
AddNode TheLordOfTheRing5 Gandalf
AddNode TheLordOfTheRing5 Samwise
AddNode TheLordOfTheRing5 Arwin

# Add six edges
AddEdge TheLordOfTheRing5 Gandalf Frodo TheHobbit
AddEdge TheLordOfTheRing5 Samwise Frodo TheReturnOfTheKing
AddEdge TheLordOfTheRing5 Samwise Frodo TheTwoTowers
AddEdge TheLordOfTheRing5 Frodo Samwise TheTwoTowers
AddEdge TheLordOfTheRing5 Samwise Arwin TheFellowshipOfTheRing
AddEdge TheLordOfTheRing5 Arwin Gandalf TheFellowshipOfTheRing

# Print the nodes in the graph and the outgoing edges from Frodo, Gandalf, Arwin, Samwise
ListNodes TheLordOfTheRing5
ListChildren TheLordOfTheRing5 Frodo
ListChildren TheLordOfTheRing5 Gandalf
ListChildren TheLordOfTheRing5 Arwin
ListChildren TheLordOfTheRing5 Samwise