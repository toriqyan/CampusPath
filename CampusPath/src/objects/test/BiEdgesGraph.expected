# Tests that a graph can be created, four nodes and
# two pairs of cross edges and two other edges can be added, 
# and four nodes and six edges can be printed out.

# Create a graph
created graph TheLordOfTheRing5

# Add four nodes
added node Frodo to TheLordOfTheRing5
added node Gandalf to TheLordOfTheRing5
added node Samwise to TheLordOfTheRing5
added node Arwin to TheLordOfTheRing5

# Add six edges
added edge TheHobbit from Gandalf to Frodo in TheLordOfTheRing5
added edge TheReturnOfTheKing from Samwise to Frodo in TheLordOfTheRing5
added edge TheTwoTowers from Samwise to Frodo in TheLordOfTheRing5
added edge TheTwoTowers from Frodo to Samwise in TheLordOfTheRing5
added edge TheFellowshipOfTheRing from Samwise to Arwin in TheLordOfTheRing5
added edge TheFellowshipOfTheRing from Arwin to Gandalf in TheLordOfTheRing5

# Print the nodes in the graph and the outgoing edges from Frodo, Gandalf, Arwin, Samwise
TheLordOfTheRing5 contains: Arwin Frodo Gandalf Samwise
the children of Frodo in TheLordOfTheRing5 are: Samwise(TheTwoTowers)
the children of Gandalf in TheLordOfTheRing5 are: Frodo(TheHobbit)
the children of Arwin in TheLordOfTheRing5 are: Gandalf(TheFellowshipOfTheRing)
the children of Samwise in TheLordOfTheRing5 are: Arwin(TheFellowshipOfTheRing) Frodo(TheReturnOfTheKing) Frodo(TheTwoTowers)